From 7b89ebd29b08df2ee80836eba1861f87cbca35de Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 29 Jan 2023 11:18:42 +0200
Subject: [PATCH] [SW-120926] gaudi3: pldm_bloader: fix pci pll div2

Since on PLDM we change the CLK driving the D2D SPI GW,
application can't reconfigure the PCI PLL over the SPI GW.
Instead, we rely on boot FSM to configure the PCI PLL, and add some
fixups in the pldm-bloader application, which runs on both dies.
Yet Boot FSM actually doesn't configure div2 of the PCI pll to
nominal clk (100MHz), so we fix it as well in the pldm-bloader.

Change-Id: Id472c33bc44d6fcd5b4a77f04d3a2cbe16da142a
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 84aee1c..2123bd6 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -167,6 +167,7 @@
 	return dev_id;
 }
 
+#define PCI_AUX_DIV_ID	2
 #define UART_DIV_ID	3
 
 /*
@@ -174,11 +175,13 @@
  * support uart prints on pldm only div3 of the pci pll should be modified. Since div3 also impacts
  * the baudrate of the d2d spi dma, pci pll of both dies should be carefully reconfigured first,
  * before using the spi gw for data transfer.
+ * since pci pll div2 is not initialized by boot fsm, it is being set here as well.
  */
 void btldr_pll_fixup(uint32_t die_id)
 {
 	gaudi3_cpu_pll_cfg[0].base = die_id ? mmD1_G4_PCI_PLL_CTRL_BASE : mmD0_G4_PCI_PLL_CTRL_BASE;
 
+	hl_pll_reconfig_div(gaudi3_cpu_pll_cfg, PCI_AUX_DIV_ID);
 	hl_pll_reconfig_div(gaudi3_cpu_pll_cfg, UART_DIV_ID);
 }
 
